Jack Voigt

In [13]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [14]:
import os
os.getcwd()

'/Users/JackVoigt/Documents/MSIA Files'

In [15]:
df1 = pd.read_csv("Census_Data_-_Selected_socioeconomic_indicators_in_Chicago__2008___2012-v2.csv")

In [16]:
pd.set_option('display.max_columns', None)
df1.head(5)

COMMUNITY_AREA_NUMBER COMMUNITY_AREA_NAME  PERCENT OF HOUSING CROWDED  \
0                    1.0         Rogers Park                         7.7   
1                    2.0          West Ridge                         7.8   
2                    3.0              Uptown                         3.8   
3                    4.0      Lincoln Square                         3.4   
4                    5.0        North Center                         0.3   

   PERCENT HOUSEHOLDS BELOW POVERTY  PERCENT AGED 16+ UNEMPLOYED  \
0                              23.6                          8.7   
1                              17.2                          8.8   
2                              24.0                          8.9   
3                              10.9                          8.2   
4                               7.5                          5.2   

   PERCENT AGED 25+ WITHOUT HIGH SCHOOL DIPLOMA  \
0                                          18.2   
1                                          20.8   
2                                          11.8   
3                                          13.4   
4                                           4.5   

   PERCENT AGED UNDER 18 OR OVER 64  PER_CAPITA_INCOME   HARDSHIP_INDEX  
0                              27.5               23939            39.0  
1                              38.5               23040            46.0  
2                              22.2               35787            20.0  
3                              25.5               37524            17.0  
4                              26.2               57123             6.0

In [17]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 78 entries, 0 to 77
Data columns (total 9 columns):
 #   Column                                        Non-Null Count  Dtype  
---  ------                                        --------------  -----  
 0   COMMUNITY_AREA_NUMBER                         77 non-null     float64
 1   COMMUNITY_AREA_NAME                           78 non-null     object 
 2   PERCENT OF HOUSING CROWDED                    78 non-null     float64
 3   PERCENT HOUSEHOLDS BELOW POVERTY              78 non-null     float64
 4   PERCENT AGED 16+ UNEMPLOYED                   78 non-null     float64
 5   PERCENT AGED 25+ WITHOUT HIGH SCHOOL DIPLOMA  78 non-null     float64
 6   PERCENT AGED UNDER 18 OR OVER 64              78 non-null     float64
 7   PER_CAPITA_INCOME                             78 non-null     int64  
 8   HARDSHIP_INDEX                                77 non-null     float64
dtypes: float64(7), int64(1), object(1)
memory usage: 5.6+ KB


In [18]:
for col in df1.columns: 
    print(col)

COMMUNITY_AREA_NUMBER
COMMUNITY_AREA_NAME
PERCENT OF HOUSING CROWDED
PERCENT HOUSEHOLDS BELOW POVERTY
PERCENT AGED 16+ UNEMPLOYED
PERCENT AGED 25+ WITHOUT HIGH SCHOOL DIPLOMA
PERCENT AGED UNDER 18 OR OVER 64
PER_CAPITA_INCOME 
HARDSHIP_INDEX


In [19]:
df1.shape

(78, 9)

In [20]:
df1.dtypes

COMMUNITY_AREA_NUMBER                           float64
COMMUNITY_AREA_NAME                              object
PERCENT OF HOUSING CROWDED                      float64
PERCENT HOUSEHOLDS BELOW POVERTY                float64
PERCENT AGED 16+ UNEMPLOYED                     float64
PERCENT AGED 25+ WITHOUT HIGH SCHOOL DIPLOMA    float64
PERCENT AGED UNDER 18 OR OVER 64                float64
PER_CAPITA_INCOME                                 int64
HARDSHIP_INDEX                                  float64
dtype: object

In [21]:
df1.isnull().sum()

COMMUNITY_AREA_NUMBER                           1
COMMUNITY_AREA_NAME                             0
PERCENT OF HOUSING CROWDED                      0
PERCENT HOUSEHOLDS BELOW POVERTY                0
PERCENT AGED 16+ UNEMPLOYED                     0
PERCENT AGED 25+ WITHOUT HIGH SCHOOL DIPLOMA    0
PERCENT AGED UNDER 18 OR OVER 64                0
PER_CAPITA_INCOME                               0
HARDSHIP_INDEX                                  1
dtype: int64

In the Chicago census database above, the dataset that includes 78 observations and 9 columns, or variables. This set gives us information about census data for different areas in Chicago. This data takes a look at the poverty and unemployment rates in these areas, educational information, and income. This dataset includes 7 columns that are numeric or float type, one column that is an object data type ("COMMUNITY_AREA_NAME"), and one column that is an integer data type ("PER_CAPITA_INCOME"). We have 1 null value in both the "COMMUNITY_AREA_NUMBER" column, and the "HARDSHIP_INDEX" column.

In [22]:
df2 = pd.read_csv("Chicago_Public_Schools_-_Progress_Report_Cards__2011-2012-v3.csv")

In [23]:
pd.set_option('display.max_columns', None)
df2.head(5)

School ID                                     NAME_OF_SCHOOL  \
0     610038                  Abraham Lincoln Elementary School   
1     610281  Adam Clayton Powell Paideia Community Academy ...   
2     610185                Adlai E Stevenson Elementary School   
3     609993                    Agustin Lara Elementary Academy   
4     610513                      Air Force Academy High School   

  Elementary, Middle, or High School          Street Address     City State  \
0                                 ES        615 W Kemper Pl   Chicago    IL   
1                                 ES  7511 S South Shore Dr   Chicago    IL   
2                                 ES     8010 S Kostner Ave   Chicago    IL   
3                                 ES     4619 S Wolcott Ave   Chicago    IL   
4                                 HS        3630 S Wells St   Chicago    IL   

   ZIP Code    Phone Number  \
0     60614  (773) 534-5720   
1     60649  (773) 535-6650   
2     60652  (773) 535-2280   
3     60609  (773) 535-4389   
4     60609  (773) 535-1590   

                                               Link   \
0  http://schoolreports.cps.edu/SchoolProgressRep...   
1  http://schoolreports.cps.edu/SchoolProgressRep...   
2  http://schoolreports.cps.edu/SchoolProgressRep...   
3  http://schoolreports.cps.edu/SchoolProgressRep...   
4  http://schoolreports.cps.edu/SchoolProgressRep...   

                      Network Manager                  Collaborative Name  \
0        Fullerton Elementary Network  NORTH-NORTHWEST SIDE COLLABORATIVE   
1           Skyway Elementary Network            SOUTH SIDE COLLABORATIVE   
2           Midway Elementary Network        SOUTHWEST SIDE COLLABORATIVE   
3         Pershing Elementary Network        SOUTHWEST SIDE COLLABORATIVE   
4  Southwest Side High School Network        SOUTHWEST SIDE COLLABORATIVE   

  Adequate Yearly Progress Made?  Track Schedule  \
0                              No       Standard   
1                              No        Track_E   
2                              No       Standard   
3                              No        Track_E   
4                             NDA       Standard   

  CPS Performance Policy Status CPS Performance Policy Level  \
0              Not on Probation                      Level 1   
1              Not on Probation                      Level 1   
2              Not on Probation                      Level 2   
3              Not on Probation                      Level 1   
4              Not on Probation              Not Enough Data   

  HEALTHY_SCHOOL_CERTIFIED Safety Icon   SAFETY_SCORE Family Involvement Icon  \
0                      Yes  Very Strong          99.0             Very Strong   
1                       No      Average          54.0                  Strong   
2                       No       Strong          61.0                     NDA   
3                       No      Average          56.0                 Average   
4                      Yes      Average          49.0                  Strong   

  Family Involvement Score Environment Icon   Environment Score  \
0                       99            Strong               74.0   
1                       66            Strong               74.0   
2                      NDA           Average               50.0   
3                       44           Average               45.0   
4                       60            Strong               60.0   

  Instruction Icon   Instruction Score Leaders Icon  Leaders Score   \
0            Strong               66.0        Strong             65   
1       Very Strong               84.0        Strong             63   
2              Weak               36.0           NDA            NDA   
3              Weak               37.0        Strong             65   
4           Average               55.0       Average             45   

  Teachers Icon  Teachers Score Parent Engagement Icon   \
0         Strong             70                  Strong   
1    

In [24]:
df2.shape

(566, 78)

In [25]:
for col in df2.columns: 
    print(col)

School ID
NAME_OF_SCHOOL
Elementary, Middle, or High School
Street Address
City
State
ZIP Code
Phone Number
Link 
Network Manager
Collaborative Name
Adequate Yearly Progress Made? 
Track Schedule
CPS Performance Policy Status
CPS Performance Policy Level
HEALTHY_SCHOOL_CERTIFIED
Safety Icon 
SAFETY_SCORE
Family Involvement Icon
Family Involvement Score
Environment Icon 
Environment Score
Instruction Icon 
Instruction Score
Leaders Icon 
Leaders Score 
Teachers Icon 
Teachers Score
Parent Engagement Icon 
Parent Engagement Score
Parent Environment Icon
Parent Environment Score
AVERAGE_STUDENT_ATTENDANCE
Rate of Misconducts (per 100 students) 
Average Teacher Attendance
Individualized Education Program Compliance Rate 
Pk-2 Literacy %
Pk-2 Math %
Gr3-5 Grade Level Math %
Gr3-5 Grade Level Read % 
Gr3-5 Keep Pace Read %
Gr3-5 Keep Pace Math %
Gr6-8 Grade Level Math %
Gr6-8 Grade Level Read %
Gr6-8 Keep Pace Math%
Gr6-8 Keep Pace Read %
Gr-8 Explore Math %
Gr-8 Explore Read %
ISAT Exceedin

In [165]:
df2.dtypes

School ID                              int64
NAME_OF_SCHOOL                        object
Elementary, Middle, or High School    object
Street Address                        object
City                                  object
                                       ...  
COMMUNITY_AREA_NUMBER                  int64
COMMUNITY_AREA_NAME                   object
Ward                                   int64
Police District                        int64
Location                              object
Length: 78, dtype: object

In [26]:
obj = df2.isnull().sum()
for key,value in obj.iteritems():
    print(key,",",value)

School ID , 0
NAME_OF_SCHOOL , 0
Elementary, Middle, or High School , 0
Street Address , 0
City , 0
State , 0
ZIP Code , 0
Phone Number , 0
Link  , 1
Network Manager , 0
Collaborative Name , 0
Adequate Yearly Progress Made?  , 0
Track Schedule , 0
CPS Performance Policy Status , 0
CPS Performance Policy Level , 0
HEALTHY_SCHOOL_CERTIFIED , 0
Safety Icon  , 0
SAFETY_SCORE , 53
Family Involvement Icon , 0
Family Involvement Score , 0
Environment Icon  , 0
Environment Score , 53
Instruction Icon  , 0
Instruction Score , 53
Leaders Icon  , 0
Leaders Score  , 0
Teachers Icon  , 0
Teachers Score , 0
Parent Engagement Icon  , 0
Parent Engagement Score , 0
Parent Environment Icon , 0
Parent Environment Score , 0
AVERAGE_STUDENT_ATTENDANCE , 1
Rate of Misconducts (per 100 students)  , 0
Average Teacher Attendance , 0
Individualized Education Program Compliance Rate  , 0
Pk-2 Literacy % , 0
Pk-2 Math % , 0
Gr3-5 Grade Level Math % , 0
Gr3-5 Grade Level Read %  , 0
Gr3-5 Keep Pace Read % , 0
Gr3-

The Chicago Public Schools data set contains data about all of the public schools in the city of Chicago. It contains 566 observations, or schools, and 78 columns. This data informs us primarily on the location of the schools, grades for the administration, student attendance and scores, and more. This dataset contains many different data types throughout its columns, and many of these variables have a large amount of null values. When we run our null value function, we can see there are many missing values in the columns "Environment Score", "Instruction Score", "ISAT Exceeding Math", "ISAT Exceeding Reading", "ISAT Exceeding Reading", "ISAT Value Add Math", and "ISAT Value Add Read". We will not use these in our database formation because of the large amount of missing values.  

We will pick 10 variables from our public schools dataset that we will use in our SQL database that are good indicators for the quality of each school. We will select, "Elementary, Middle, or High School", "HEALTHY_SCHOOL_CERTIFIED", "SAFETY_SCORE", "Environment Score", "Teachers Score", "Parent Engagement Score", "AVERAGE_STUDENT_ATTENDANCE", "Rate of misconducts (per 100 students)", "Average Teacher Attendance", "Individualized Education Program Compliance Rate", and "COLLEGE_ENROLLMENT".  We will now clean the data and drop undesired columns from our data.

In [27]:
df2.drop(labels = ["Phone Number", "Link ", "Network Manager", "Collaborative Name",
                  "Adequate Yearly Progress Made? ", "Track Schedule", "CPS Performance Policy Status", "CPS Performance Policy Level", "Safety Icon ", 
                  "Family Involvement Icon", "Family Involvement Score", "Environment Icon ", "Instruction Icon ",
                  "Instruction Score", "Leaders Icon ", "Leaders Score ", "Teachers Icon ", "Parent Engagement Icon ",
                 "Parent Environment Icon", "Parent Environment Score", "Pk-2 Literacy %",
                  "Pk-2 Math %", "Gr3-5 Grade Level Math %", "Gr3-5 Grade Level Read % ", "Gr3-5 Keep Pace Read %", 
                  "Gr3-5 Keep Pace Math %", "Gr6-8 Grade Level Math %", "Gr6-8 Grade Level Read %", "Gr6-8 Keep Pace Math%",
                  "Gr6-8 Keep Pace Read %", "Gr-8 Explore Math %", "Gr-8 Explore Read %", "ISAT Exceeding Math %", 
                  "ISAT Exceeding Reading % ", "ISAT Value Add Math", "ISAT Value Add Read", "ISAT Value Add Color Math",
                  "ISAT Value Add Color Read", "Students Taking  Algebra %", "Students Passing  Algebra %", "9th Grade EXPLORE (2009) ",
                  "9th Grade EXPLORE (2010) ", "10th Grade PLAN (2009) ", "10th Grade PLAN (2010) ", "Net Change EXPLORE and PLAN",
                  "11th Grade Average ACT (2011) ", "Net Change PLAN and ACT", "College Eligibility %", "Graduation Rate %",
                  "College Enrollment Rate %", "General Services Route ", "Freshman on Track Rate %", "X_COORDINATE", "Y_COORDINATE",
                  "Latitude", "Longitude", "COMMUNITY_AREA_NUMBER", "COMMUNITY_AREA_NAME", "Ward", "Police District", "Location"], inplace = True, axis = 1)

In [29]:
df2.head(5)

School ID                                     NAME_OF_SCHOOL  \
0     610038                  Abraham Lincoln Elementary School   
1     610281  Adam Clayton Powell Paideia Community Academy ...   
2     610185                Adlai E Stevenson Elementary School   
3     609993                    Agustin Lara Elementary Academy   
4     610513                      Air Force Academy High School   

  Elementary, Middle, or High School          Street Address     City State  \
0                                 ES        615 W Kemper Pl   Chicago    IL   
1                                 ES  7511 S South Shore Dr   Chicago    IL   
2                                 ES     8010 S Kostner Ave   Chicago    IL   
3                                 ES     4619 S Wolcott Ave   Chicago    IL   
4                                 HS        3630 S Wells St   Chicago    IL   

   ZIP Code HEALTHY_SCHOOL_CERTIFIED  SAFETY_SCORE  Environment Score  \
0     60614                      Yes          99.0               74.0   
1     60649                       No          54.0               74.0   
2     60652                       No          61.0               50.0   
3     60609                       No          56.0               45.0   
4     60609                      Yes          49.0               60.0   

  Teachers Score Parent Engagement Score AVERAGE_STUDENT_ATTENDANCE  \
0             70                      56                     96.00%   
1             76                      46                     95.60%   
2            NDA                      47                     95.70%   
3             48                      53                     95.50%   
4             54                      53                     93.30%   

   Rate of Misconducts (per 100 students)  Average Teacher Attendance  \
0                                      2.0                     96.40%   
1                                     15.7                     95.30%   
2                                      2.3                     94.70%   
3                                     10.4                     95.80%   
4                                     15.6                     96.90%   

  Individualized Education Program Compliance Rate   COLLEGE_ENROLLMENT  
0                                            95.80%                 813  
1                                           100.00%                 521  
2                                            98.30%                1324  
3                                           100.00%                 556  
4                                           100.00%                 302

Above is our updated dataframe. We have dropped all of our columns that are not relevant or good indicators of these schools. The best choice for missing data here is listwise deletion, because our dataset will still include many rows of data, so our analysis can still be accurate, and many of the observations that have mising values in one column, have missing values in one or more other columns. So we will apply listwise deletion here. 

In [30]:
df2.drop(df2[df2['Teachers Score'] == "NDA"].index, inplace = True)

In [31]:
df2.head()

School ID                                     NAME_OF_SCHOOL  \
0     610038                  Abraham Lincoln Elementary School   
1     610281  Adam Clayton Powell Paideia Community Academy ...   
3     609993                    Agustin Lara Elementary Academy   
4     610513                      Air Force Academy High School   
5     610212                  Albany Park Multicultural Academy   

  Elementary, Middle, or High School          Street Address     City State  \
0                                 ES        615 W Kemper Pl   Chicago    IL   
1                                 ES  7511 S South Shore Dr   Chicago    IL   
3                                 ES     4619 S Wolcott Ave   Chicago    IL   
4                                 HS        3630 S Wells St   Chicago    IL   
5                                 MS      4929 N Sawyer Ave   Chicago    IL   

   ZIP Code HEALTHY_SCHOOL_CERTIFIED  SAFETY_SCORE  Environment Score  \
0     60614                      Yes          99.0               74.0   
1     60649                       No          54.0               74.0   
3     60609                       No          56.0               45.0   
4     60609                      Yes          49.0               60.0   
5     60625                       No          66.0               66.0   

  Teachers Score Parent Engagement Score AVERAGE_STUDENT_ATTENDANCE  \
0             70                      56                     96.00%   
1             76                      46                     95.60%   
3             48                      53                     95.50%   
4             54                      53                     93.30%   
5             50                      46                     97.00%   

   Rate of Misconducts (per 100 students)  Average Teacher Attendance  \
0                                      2.0                     96.40%   
1                                     15.7                     95.30%   
3                                     10.4                     95.80%   
4                                     15.6                     96.90%   
5                                      2.3                     96.90%   

  Individualized Education Program Compliance Rate   COLLEGE_ENROLLMENT  
0                                            95.80%                 813  
1                                           100.00%                 521  
3                                           100.00%                 556  
4                                           100.00%                 302  
5                                           100.00%                 266

In [32]:
df2.shape

(295, 17)

After our deletion of rows with missing values and the removal of unwanted or irrelevant columns, we know have a dataset on Chicago public schools data that has 295 observations and 17 columns. We will now create an SQlite database, and create data tables for each of our data sets.

In [36]:
from pathlib import Path
Path("df2.db").touch()

In [45]:
import sqlite3

In [73]:
conn=sqlite3.connect('/Users/JackVoigt/Documents/MSIA Files/Census_CD_Database.sqlite')

In [74]:
cur=conn.cursor()

In [75]:
 cur.execute('''CREATE TABLE IF NOT EXISTS CD (COMMUNITY_AREA_NUMBER FLOAT PRIMARY KEY,
               [COMMUNITY_AREA_NAME] VARCHAR,[PERCENT OF HOUSING CROWDED] FLOAT,
               [PERCENT HOUSEHOLDS BELOW POVERTY] FLOAT,[PERCENT AGED 16+ UNEMPLOYED] FLOAT,
               [PERCENT AGED 25+ WITHOUT HIGH SCHOOL DIPLOMA] FLOAT, [PERCENT AGED UNDER 18 OR OVER 64] FLOAT,
               [PER_CAPITA_INCOME] FLOAT, 
               [HARDSHIP_INDEX] FLOAT)''')
    
conn.close()

In [76]:
#Create connection 

conn=sqlite3.connect('/Users/JackVoigt/Documents/MSIA Files/Census_CD_Database.sqlite')


cur=conn.cursor()

#transfer data 
df1.to_sql('CD',conn, if_exists='replace', index=False)


conn.close()

In [77]:
conn2=sqlite3.connect('/Users/JackVoigt/Documents/MSIA Files/Census_CPS_Database.sqlite')

In [78]:
cur2=conn2.cursor()

In [79]:
 cur2.execute('''CREATE TABLE IF NOT EXISTS CPS (School ID INT PRIMARY KEY,
               [NAME_OF_SCHOOL] VARCHAR,[Elementary, Middle, or High School] VARCHAR,
               [Street Address] VARCHAR,[City] VARCHAR,
               [State] VARCHAR, [ZIP Code] INT,
               [HEALTHY_SCHOOL_CERTIFIED] VARCHAR, [SAFETY_SCORE] FLOAT, [Environment Score] FLOAT,
               [Teachers Score] FLOAT, [Parent Engagement Score] FLOAT, [AVERAGE_STUDENT_ATTENDANCE] FLOAT,
               [Rate of Misconducts (per 100 students)] FLOAT, [Average Teacher Attendance] FLOAT, 
               [Individualized Education Program Compliance Rate] FLOAT, 
               [COLLEGE_ENROLLMENT] INT)''')
    
conn2.close()

In [80]:
#Create connection 

conn2=sqlite3.connect('/Users/JackVoigt/Documents/MSIA Files/Census_CPS_Database.sqlite')


cur2=conn2.cursor()

#transfer data 
df2.to_sql('CPS',conn2, if_exists='replace', index=False)


conn2.close()